pkg/cloudprovider/aws/instance.go

```diff
@@ -143,6 +197,13 @@ func (p *InstanceProvider) launchInstances(ctx context.Context, constraints *v1a
 	if err != nil {
 		return nil, fmt.Errorf("creating fleet %w", err)
 	}
+	if len(createFleetOutput.Errors) > 0 {
+		for _, fleetError := range createFleetOutput.Errors {
+			if isInsufficientCapacityCode(*fleetError.ErrorCode) {
```

Comment:
use aws.StringValue(fleetError.ErrorCode) to avoid NPE if AWS ever serves us a bad response.